src/components/data/RenameTableModal.js

```diff
@@ -0,0 +1,56 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import { ButtonPrimary, IdContainer, spinnerOverlay } from 'src/components/common'
+import { ValidatedInput } from 'src/components/input'
+import Modal from 'src/components/Modal'
+import { Ajax } from 'src/libs/ajax'
+import { withErrorReporting } from 'src/libs/error'
+import Events from 'src/libs/events'
+import { FormLabel } from 'src/libs/forms'
+import * as Utils from 'src/libs/utils'
+
+
+export const tableNameInput = ({ inputProps, ...props }) => h(ValidatedInput, {
+  ...props,
+  inputProps: {
+    ...inputProps,
+    autoFocus: true,
+    placeholder: 'Enter a name'
+  }
+})
+
+const RenameTableModal = ({ onDismiss, onUpdateSuccess, namespace, name, selectedDataType }) => {
+  // State
+  const [newName, setNewName] = useState('')
+  const [renaming, setRenaming] = useState(false)
+
+  return h(Modal, {
+    onDismiss,
+    title: 'Rename Data Table',
+    okButton: h(ButtonPrimary, {
+      disabled: renaming,
+      onClick: _.flow(
+        withErrorReporting('Error renaming data table.'),
+        Utils.withBusyState(setRenaming)
+      )(async () => {
+        await Ajax().Metrics.captureEvent(Events.workspaceDataRenameTable, { oldName: selectedDataType, newName })
```

Comment:
This will record an event even if the rename fails. This is fine as long as it's understood that the event represents an _intent_ to rename a data table. My guess is that the most common errors would be an invalid or duplicate table name?

Reply:
That's a good question. I was thinking it'd be good to capture intent but I will double-check with those who have more of a stake in these metrics than I do

And yes, those would be the typical failure reasons

Reply:
Chiming in here on behalf of the metrics, we added these to capture intent to see commonly used operations. +1 the current implementation, we could refine/debate some edgecases here, but it's over-optimizing for how we're using the metrics right now (just using them as a rough guide for what gets clicked)